server/routes/contacts/manage/contact-search/contactSearchSchema.test.ts

```diff
@@ -0,0 +1,211 @@
+        lastName: ["Contact's last name must not contain special characters"],
+      })
+    })
+
+    it('should not pass validation when dob is not valid', async () => {
+      // Given
+      const form = { ...baseForm, lastName: 'testname', day: 32, month: 13, year: 100 }
+
+      // When
+      const result = await doValidate(form)
+
+      // Then
+      expect(result.success).toStrictEqual(false)
+      const deduplicatedFieldErrors = deduplicateFieldErrors(result)
+      expect(deduplicatedFieldErrors).toStrictEqual({
+        day: ['Enter a valid day of the month (1-31)'],
+        month: ['Enter a valid month (1-12)'],
+        year: ['Enter a valid year. Must be at least 1900'],
+      })
+    })
+
+    it('should not pass validation when dob is in the future', async () => {
+      // Given
+      const date = new Date(Date.now())
+      date.setDate(date.getDate() + 1)
+
+      const form = { ...baseForm, lastName: 'testname', day: 1, month: 12, year: date.getFullYear() }
+
+      // When
+      const result = await doValidate(form)
+
+      // Then
+      expect(result.success).toStrictEqual(false)
+      const deduplicatedFieldErrors = deduplicateFieldErrors(result)
+      expect(deduplicatedFieldErrors).toStrictEqual({
+        dob: ['The date of birth must not be in the future'],
+      })
+    })
+
+    it.each(['mum@example.com', "someone (don't know who)", 'Said £10 to tell me', '* look this up later', '911'])(
+      'names should be limited to valid name chars',
+      async (invalidName: string) => {
+        // Given
+        const form = {
+          firstName: invalidName,
+          middleName: invalidName,
+          lastName: invalidName,
+        }
+
+        // When
+        const result = await doValidate(form)
+
+        // Then
+        expect(result.success).toStrictEqual(false)
+        const deduplicatedFieldErrors = deduplicateFieldErrors(result)
+        expect(deduplicatedFieldErrors).toStrictEqual({
+          firstName: [`Contact's first name must not contain special characters`],
+          lastName: [`Contact's last name must not contain special characters`],
+          middleName: [`Contact's middle name must not contain special characters`],
+        })
+      },
+    )
+
+    it.each(['Foo jr.', 'Bar, the III', 'foo-bar', "Foo Mc'Bar", 'Foo Bar'])(
+      'some special chars are valid in names',
+      async (validName: string) => {
+        // Given
+        const form = {
+          firstName: validName,
+          middleName: validName,
+          lastName: validName,
+        }
+
+        // When
+        const result = await doValidate(form)
+
+        // Then
+        expect(result.success).toStrictEqual(true)
+      },
+    )
+
+    const doValidate = async (form: Form) => {
+      const schema = await contactSearchSchema()()
+      return schema.safeParse(form)
+    }
+  })
+
+  describe('contactSearchSchema', () => {
+    type Form = {
```

Comment:
2 x name 'contactSearchSchema' blocks?

Reply:
Maybe one should be `Validate names` and the other `Validate date of birth`?

Reply:
describe block renamed